### ISSUE TITLE
elixir-koans indexes extremely slowly

### ISSUE THREAD

**Issue by wizonesolutions** [KronicDeth/intellij-elixir#1006]
When I turn off the Elixir plugin and restart, it indexes quickly.

This is what I did:

1. `git clone https://github.com/elixirkoans/elixir-koans`
1. `cd elixir-koans; mix deps.get`
1. (IntelliJ IDEA Ultimate) New -> Project From Existing Sources...
1. Choose the elixir-koans directory
1. Say that it's a Mix project
1. Import elixir-koans

Expected: Indexing happens within 30-60s or so, like with most new projects.

Actual: Indexing does progress, but changing files will cause the IDE to become unresponsive for 2-3 minutes.

I tested this with and without Elixir Koans itself running (I thought maybe there was some crazy inotify contention going on, but that doesn't seem to be the case).

I haven't figured out a workaround yet, other than turning off the Elixir plugin...but I don't really want to, obviously :)

**Comment by wizonesolutions** [comment]
Note: Checked that the Credo inspection isn't randomly enabled. I am also seeing this in a separate, private repo (which is weird, because this repo worked fine just the other day). I dunno, let me try restarting my computer.

**Comment by wizonesolutions** [comment]
(restarting my computer didn't help)

**Comment by wizonesolutions** [comment]
More info! I tried the plugin in PhpStorm and noticed it was fast, and that it was prompting me to define the Elixir Module SDK.

I unset it in IntelliJ IDEA and indexing went much faster.

I'm not using the latest Elixir in this project; I did `brew install brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/9c27c28af699bcaa0245617e486664c80fd47d2e/Formula/elixir.rb` (Elixir 1.4.2) on my Mac because I needed Elixir 1.4, and I couldn't get this plugin to recognize `asdf` or `kiex`-installed Elixirs.

I assume something between the interaction of Elixir 1.4.2, my `kerl`-installed Erlang OTP 19 (which the plugin did pick up OK) is the issue. No clue what, but I think this setup will allow you to see the issue.
